Query trait bounds have changed, breaks build

Towards Testnet 60 (https://github.com/penumbra-zone/penumbra/issues/2984), we just merged https://github.com/penumbra-zone/penumbra/pull/2954, which breaks the Osiris build. Reverting Penumbra deps to prior to that PR, it compiles again. See example failing build here: https://github.com/penumbra-zone/osiris/actions/runs/6150328208/job/16688046979#step:7:432

```
error[E0599]: the method `liquidity_positions_by_price` exists for struct `SpecificQueryServiceClient<Channel>`, but its trait bounds were not satisfied
   --> src/trader.rs:602:20
    |
602 |             client.liquidity_positions_by_price(LiquidityPositionsByPriceRequest {
    |             -------^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method cannot be called on `SpecificQueryServiceClient<Channel>` due to unsatisfied trait bounds
    |
   ::: /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tonic-0.8.3/src/transport/channel/mod.rs:69:1
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,7 +16,9 @@ penumbra-keys = { path = "../penumbra/crates/core/keys" }
 penumbra-dex = { path = "../penumbra/crates/core/component/dex" }
 penumbra-custody = { path = "../penumbra/crates/custody" }
 penumbra-view = { path = "../penumbra/crates/view" }
-penumbra-transaction = { path = "../penumbra/crates/core/transaction", features = ["download-proving-keys"] }
+penumbra-transaction = { path = "../penumbra/crates/core/transaction", features = [
+    "download-proving-keys",
+] }
 
 # External dependencies
 anyhow = "1"
@@ -34,7 +36,7 @@ rand = "0.8"
 reqwest = "0.11"
 indexmap = "1.8"
 chrono = { version = "0.4", features = ["serde"] }
-tonic = { version = "0.8.1", features = ["tls-webpki-roots", "tls"] }
-serde = { version = "1", features = ["derive"]}
+tonic = { version = "0.9", features = ["tls-webpki-roots", "tls"] }
+serde = { version = "1", features = ["derive"] }
 binance = { git = "https://github.com/wisespace-io/binance-rs.git" }
 url = "2"

